src/modules/api/user/link/link.controller.ts

```diff
@@ -1,11 +1,39 @@
-import { IResult } from './../../../../common/interfaces/response';
-import { Controller, Post, Body, Param } from '@nestjs/common';
 import { LinkService } from './link.service';
+import { IResult } from './../../../../common/interfaces/response';
+import { Controller, Get, Post, Body, Param } from '@nestjs/common';
 import { LinkDTO } from './link.dto';
 
 @Controller('users/:userId/links')
 export class LinkController {
   constructor(private readonly linkService: LinkService) {}
+  @Get()
+  async getAllLinksForUserId(
+    @Param('userId') userId: string,
+  ): Promise<IResult<LinkDTO>> {
+    try {
+      const links = await this.linkService.getAllLinksOfUser(userId);
+      if (links) {
+        const linkArray = links.map(link => {
+          return {
+            name: link.name,
```

Comment:
We can still get rid of the `return` word by using single line arrow notation :sweat_smile:

Reply:
Like are you asking me to just remove the return?

Reply:
Yeah, and the extra curly braces.

```
const linkArray = links.map(link => {
            name: link.name,
             customUrl: link.customUrl,
            redirectsTo: link.redirectsTo,
            status: link.status,
            inPool: link.inPool
          } as LinkDTO
);
```
should work...

Reply:
const linkArray = links.map(link =>{
            name: link.name,
            customUrl: link.customUrl,
            redirectsTo: link.redirectsTo,
            status: link.status,
            inPool: link.inPool,
        } as LinkDTO);
this was not working so i added that return statement as well

Reply:
Okay, it doesn't work with returning objects I guess.

Reply:
you have to change like this
.map(link=>({
}));

Reply:
May be

Reply:
() is default return of arrow function.